src/components/Search/ResultsList.tsx

```diff
@@ -18,26 +19,31 @@ function ResultsList({
   results,
   selectedResultId,
   isMobile = false,
+  paging,
 }: ResultsListProps) {
   return (
     <>
-      {results.map((result) => (
-        <div
-          className={
-            isMobile
-              ? MOBILE_CLASSES
-              : selectedResultId === result.id
-              ? DESKTOP_CLASSES_ACTIVE
-              : DESKTOP_CLASSES
-          }
-          // Only set id in the desktop list to avoid creating
-          // duplicate DOM elements with same id in mobile list
-          id={isMobile ? undefined : result.id}
-          key={result.id}
-        >
-          <ResultCard data={result} />
-        </div>
-      ))}
+      {results.length > 0 &&
```

Comment:
instead of passing down the pagination props into the list thru the `*Results` components, what if we used a context to track pagination state, and then the result list component and consume that context state (as can any other components that need to know about current page or page size). 

See the CompareContext for an example, which is in `Search.tsx` (but not on your branch, you need to update :) )

Reply:
Great suggestion as I was stumped on this, I am taking a look at the CompareContext for reference, thanks!